test/e2e/monitoring_stack_controller_test.go

```diff
@@ -355,6 +359,20 @@ func assertAlertmanagerReceivesAlerts(t *testing.T) {
 	}
 }
 
+func cleanupGrafanaDS(t *testing.T) {
+	ms := newMonitoringStack(t, "test-cleanup-grafana-ds")
+	datasourceName := msctrl.GrafanaDSName(ms)
+
+	err := f.K8sClient.Create(context.Background(), ms)
+	assert.NilError(t, err, "failed to create a monitoring stack")
+
+	grafanaDS := &grafanav1alpha1.GrafanaDataSource{}
+	f.AssertResourceEventuallyExists(datasourceName, goctrl.Namespace, grafanaDS)(t)
+
```

Comment:
We can add another validation here that the finalizer is set on the `ms` that was created above